ros_ws/src/autonomous/emergency_stop/src/emergency_stop.cpp

```diff
@@ -2,54 +2,114 @@
 #include <std_msgs/Time.h>
 
 EmergencyStop::EmergencyStop()
+    : m_emergency_status(EmergencyStatus::UNUSED)
+    , m_debug_geometry(this->m_node_handle, TOPIC_VISUALIZATION, LIDAR_FRAME)
 {
-    m_lidar_subscriber =
-        m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &EmergencyStop::lidarCallback, this);
-    m_emergency_stop_publisher = m_node_handle.advertise<std_msgs::Time>(TOPIC_EMERGENCY_STOP, 1);
+    this->m_lidar_subscriber =
+        this->m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &EmergencyStop::lidarCallback, this);
+    this->m_emergency_stop_publisher = this->m_node_handle.advertise<std_msgs::Time>(TOPIC_EMERGENCY_STOP, 1);
+    this->configureParameters();
+}
+
+void EmergencyStop::configureParameters()
+{
+    ros::NodeHandle private_node_handle("~");
+
+    private_node_handle.getParam(RANGE_THRESHOLD, this->m_range_threshold);
+    if ((this->m_range_threshold > 2.f) || (this->m_range_threshold < 0.3f))
```

Comment:
If 2 and 0.3 should be excluded, then the `>` should be `>=` and vice versa. But wouldn't it be fine if they were both included?
```suggestion
    if ((this->m_range_threshold >= 2.f) || (this->m_range_threshold <= 0.3f))
```

Reply:
Yes. Regarding your question, i think it would be fine to include 2, but 0.3 is so close to the car, it wouldn't really make sense as a min value for the emergency stop.